spec/rest/email_validations_spec.rb

```diff
@@ -126,6 +126,28 @@
         end
 
       end
+
+      context "will refresh data - on consecutive calls" do 
+        let(:incompleted_query) do
+          { "progress"=> { "noOfTotalEntries" => 0, "noOfCompletedEntries" => 1 } }
+        end
+
+        let(:completed_query) do
+          { "progress"=> { "noOfTotalEntries" => 1, "noOfCompletedEntries" => 1 } }
+        end
+
+        it "calls query API on current object if job is not finished in first call" do
+          allow(@email_validations).to receive(:query).and_return(incompleted_query)
+          @email_validations.query
+          expect(@email_validations.instance_variable_set('@response', incompleted_query))
+          expect(@email_validations.completed?).to be false
+
+          allow(@email_validations).to receive(:query).and_return(completed_query)
+          @email_validations.query
+          expect(@email_validations.instance_variable_set('@response', completed_query))
+          expect(@email_validations.completed?).to be true
+        end
```

Comment:
I don't agree with this test implementation. 

In the first line you are mocking the `query` method and on the second you are calling the `query` method, which will not execute the real method.  So, this test is not testing the changed line `unless is_job_completed?` on the `query` method. 

I also don't understand what is the purpose of this expectation:
```
expect(@email_validations.instance_variable_set('@response', incompleted_query))
```

Your test name is `calls query API` and you don't test if the API is being called. 

I think this tests should look like this (didn't tested it):
```
it "should call the api when not completed" do
  @email_validations.instance_variable_set('@response', incompleted_query)
  request = double()
  expect(resource).to receive(:[]).with('fake').and_return(request)
  expect(request).to receive(:get).with(content_type: :json).and_return(incompleted_query.to_json)
  expect(JSON).to receive(:parse)
  expect(@email_validations.completed?).to be false
end

it "should verify if completed after update results from the api" do
  @email_validations.instance_variable_set('@response', incompleted_query)
  request = double()
  expect(resource).to receive(:[]).with('fake').and_return(request)
  expect(request).to receive(:get).with(content_type: :json).and_return(completed_query.to_json)
  expect(JSON).to receive(:parse)
  expect(@email_validations.completed?).to be true
end

it "should not call the api when completed" do
  @email_validations.instance_variable_set('@response', completed_query)
  expect(resource).to_not receive(:[])
  expect(@email_validations.completed?).to be true
end

```